xyzservices/lib.py

```diff
@@ -81,6 +81,42 @@ def _repr_html_(self, inside=False):
 
         return html
 
+    def flatten(self) -> "Bunch":
+        """Return the nested :class:`Bunch` collapsed into one level.
+
+        Note: Formerly nested :class:`TileProvider`s cannot be accessed as attributes
+        and have to be accessed via index operator (as in ``flat["OpenStreetMap.Mapnik"]
+        not ``flat.OpenStreetMap.Mapnik``).
```

Comment:
We use the same name of TileProvider in the nested and flat versions, which breaks attribute access. I don't think there's a way around that though. Considering the use case of this function, it may as well be okay like this.

Reply:
Yes, I think that's OK here

Reply:
I've actually changed it in the end to return pure dict of TileProviders instead of a Bunch I had there initially to avoid this confusion.

Reply:
Ah, yes, good idea